lib/promiscuous_black_hole/db.rb

```diff
@@ -4,16 +4,8 @@ def self.connection
       @@connection ||= Sequel.postgres(Config.connection_args.merge(:max_connections => 10))
     end
 
-    def self.table_exists?(table)
-      exists = connection[<<-sql]
-        SELECT EXISTS (
-          SELECT 1
-          FROM   information_schema.tables
-          WHERE  table_name = '#{table}'
-          AND    table_schema = ANY (CURRENT_SCHEMAS(false))
-        );
-      sql
-      exists.first[:exists]
+    def self.table_exists?(table, opts = {})
+      DB.tables(opts).include?(table.to_sym)
```

Comment:
i did a quick benchmark, and this implementation is much cleaner without being any slower